src/main.c

```diff
@@ -251,8 +255,9 @@ int enumerate_namespaces_for_controller(struct nvme_controller *ctrl)
         snprintf(namespace_path, sizeof(namespace_path), "/dev/%s", namelist[i]->d_name);
 
         char *vs = query_namespace_vs(namespace_path);
-        printf("%s: %s\n", namespace_path, vs);
-        free(vs);
+        if (vs != NULL) {
+            free(vs);
```

Comment:
dont need this check, free works with null pointer

Reply:
fair point, will update.

will also improve some other perrors in similar manner too